### ISSUE TITLE
Project group resource is always created in default space

### ISSUE THREAD

**Issue by danielllek** [OctopusDeployLabs/terraform-provider-octopusdeploy#475]
**Describe the bug**
Project group resource is always created in default space.

**Steps to reproduce**
```
resource "octopusdeploy_project_group" "df_iac" {
  name        = "DF IaC"
  space_id    = "Spaces-122"
}
```
$ terraform apply
```
Terraform will perform the following actions:

  # octopusdeploy_project_group.df_iac will be created
  + resource "octopusdeploy_project_group" "df_iac" {
      + description         = (known after apply)
      + id                  = (known after apply)
      + name                = "DF IaC"
      + retention_policy_id = (known after apply)
      + space_id            = "Spaces-122"
    }
```
next terraform applies (it never gets into Spaces-122, always this change is shown):
```
Terraform will perform the following actions:

  # octopusdeploy_project_group.df_iac will be updated in-place
  ~ resource "octopusdeploy_project_group" "df_iac" {
        id       = "ProjectGroups-584"
        name     = "DF IaC"
      ~ space_id = "Spaces-1" -> "Spaces-122"
    }
```

**Expected behavior**
It should be created in the given space.

**Environment and versions:**
- OS: `Linux`
- Octopus Server Version: `2021.3.8275`
- Terraform Version: `v1.3.3`
- Octopus Terraform Provider Version: `0.10.1`

**Comment by danielllek** [comment]
I have similar problem with Project, it never gets created in my space.
What am I doing wrong? :-)

```
  # octopusdeploy_project.df_iac will be updated in-place
  ~ resource "octopusdeploy_project" "df_iac" {
        id                                   = "Projects-1346"
        name                                 = "DF IaC"
      ~ space_id                             = "Spaces-1" -> "Spaces-122"
        # (14 unchanged attributes hidden)

        # (3 unchanged blocks hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

**Comment by slewis74** [comment]
Hi @danielllek, this is unfortunately a known issue/limitation, in that the `space_id` on the resource gets overridden/ignored. This is a side-effect of the way Octopus' API and routes work, the route takes priority over what's in the resource, and the way the routes are currently being handled in Terraform is that they come from the provider.
I.e. if you set `space_id` on the provider, where you set the `address` and `api_key`, then it should put the resources in the correct space.
Apologies for the inconvenience. This is on our radar, and we have plans for a major update to the Go API client and the Terraform provider to address this issue, but unsure on when that will get scheduled.

**Comment by danielllek** [comment]
@slewis74 Thanks for the info, I'll use space_id in provider.

 I couldn't find this info in terraform provider docs - maybe you could add the info there until its fixed? :)

**Comment by domenicsim1** [comment]
Fixed in #558 
Released in provider version: v0.13
